src/vehicle.cpp

```diff
@@ -5323,6 +5338,29 @@ void vehicle::gain_moves()
     }
 }
 
+bool vehicle::decrement_summon_timer()
+{
+    if( !summon_time_limit ) {
+        return false;
+    }
+    if( *summon_time_limit <= 0_turns ) {
+        for( auto &p : parts ) {
+            for( const auto &e : p.items ) {
+                g->m.add_item_or_charges( global_part_pos3( p ), e );
+            }
+            p.items.clear();
+        }
```

Comment:
reading through this nested for loop, shouldn't this be a function like dump_items() or something?